src/webapp/components/ClusterMapLayers.jsx

```diff
@@ -24,9 +24,29 @@ const handleClusterOnClick = (map, e) => {
     });
 };
 
-const ClusterMapLayers = () => {
+const makePopupData = (features, lngLat) => {
+  return features.map(feat => {
+    const metaSrc = feat.properties.meta;
+    const meta =
+      typeof metaSrc == "string" ? JSON.parse(feat.properties.meta) : metaSrc;
+    return {
+      lngLat,
+      meta
+    };
+  });
+};
+
+const ClusterMapLayers = ({ geoJsonData, paramRequest }) => {
   const [popup, setPopup] = useState();
 
+  // display popup if request code is present in URL search param
+  useEffect(() => {
```

Comment:
why do we need useEffect? don't we have all the info on render if paramRequest is truthy?

Reply:
I'm not 100% sure on the details of this but if you remove the useEffect wrapper, React just tries to rerender the component over and over again, resulting in `Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.` I think it's because we are updating the state during render, versus after render if we useEffect. EIther way when working with maps, I have found it to be best to let layers render first before imposing state on it to prevent an unexpected sequence of events.